docs/api-reference/sql-api.md

```diff
@@ -23,167 +23,287 @@ sidebar_label: Druid SQL
 
-To use the SQL API to make Druid SQL queries, send your query to the Router using the POST method:
-```
-POST https://ROUTER:8888/druid/v2/sql/
-```  
+### Submit a query
 
-Submit your query as the value of a "query" field in the JSON object within the request payload. For example:
-```json
-{"query" : "SELECT COUNT(*) FROM data_source WHERE foo = 'bar'"}
-```
+Submit a SQL-based query in the JSON request body. Returns a JSON object with the database results and a set of header metadata associated with the query. 
 
-### Request body
-      
-|Property|Description|Default|
-|--------|----|-----------|
-|`query`|SQL query string.| none (required)|
-|`resultFormat`|Format of query results. See [Responses](#responses) for details.|`"object"`|
-|`header`|Whether or not to include a header row for the query result. See [Responses](#responses) for details.|`false`|
-|`typesHeader`|Whether or not to include type information in the header. Can only be set when `header` is also `true`. See [Responses](#responses) for details.|`false`|
-|`sqlTypesHeader`|Whether or not to include SQL type information in the header. Can only be set when `header` is also `true`. See [Responses](#responses) for details.|`false`|
-|`context`|JSON object containing [SQL query context parameters](../querying/sql-query-context.md).|`{}` (empty)|
-|`parameters`|List of query parameters for parameterized queries. Each parameter in the list should be a JSON object like `{"type": "VARCHAR", "value": "foo"}`. The type should be a SQL type; see [Data types](../querying/sql-data-types.md) for a list of supported SQL types.|`[]` (empty)|
-
-You can use _curl_ to send SQL queries from the command-line:
-
-```bash
-$ cat query.json
-{"query":"SELECT COUNT(*) AS TheCount FROM data_source"}
-
-$ curl -XPOST -H'Content-Type: application/json' http://ROUTER:8888/druid/v2/sql/ -d @query.json
-[{"TheCount":24433}]
-```
+This endpoint also supports querying metadata by using [metadata tables](../querying/sql-metadata-tables.md).
+
+#### URL
+
+<code class="postAPI">POST</code> <code>/druid/v2/sql</code>
+
+#### Request body
+
+* `query`: SQL query string.
+* `resultFormat`: Format of query results.
+  * `object`: Returns a JSON array of JSON objects with the HTTP header `Content-Type: application/json`.
+  * `array`: Returns a JSON array of JSON arrays with the HTTP header `Content-Type: application/json`.
+  * `objectLines`: Returns newline-delimited JSON objects with a trailing blank line. Sent with the HTTP header `Content-Type: text/plain`.
+  * `arrayLines`: Returns newline-delimited JSON arrays with a trailing blank line. Sent with the HTTP header `Content-Type: text/plain`. 
+  * `csv`: Returns a comma-separated values with one row per line and a trailing blank line. Sent with the HTTP header `Content-Type: text/csv`. 
+* `header`: Adds a header row with information on column names for the query result when set to `true`. You can optionally include `typesHeader` and `sqlTypesHeader`. 
```

Comment:
What do you think about including example snippets for when these Boolean values are set to true? Something like:

### for the header description
`````
When `header` is `true`, Druid includes a row similar to the following:
* For an `object` or `objectLines` result format:
   ```json
   <EXAMPLE>
   ```

* For an `array`, `arrayLines`, or `csv` result format:
   ```json
   <EXAMPLE>
   ```
`````

### for the typesHeader description
`````
When `header` is `true` and `typesHeader` is true, Druid includes a row similar to the following:
* For an `object` or `objectLines` result format:
   ```json
   <EXAMPLE>
   ```

* For an `array`, `arrayLines`, or `csv` result format:
   ```json
   <EXAMPLE>
   ```
`````

And then also something similar for sqlTypesHeader.

That way we can capture some of the information from the results format table:
> If header is true, the first row is an object where the fields are column names. Each field's value is either null (if typesHeader and sqlTypesHeader are false) or an object that contains the Druid type as type (if typesHeader is true) and the SQL type as sqlType (if sqlTypesHeader is true).

Reply:
Though this can get unwieldy. Consider other ways to format and present this information too.